### ISSUE TITLE
Payment log entries are not being saved for refunds

### ISSUE THREAD

**Issue by kennyadsl** [solidusio/solidus#4866]
Apparently, refunding a payment is not creating a payment log entry when executed. 

![Screenshot 2023-01-23 at 16 56 05@2x](https://user-images.githubusercontent.com/167946/214086156-a9407fa3-9d01-48ee-a1ba-974010c42bd4.png)

☝️ In this example, the order has been cancelled, and a payment have been automatically refunded with the `credit` method on the payment method/gateway. This is not attaching the log entry event to the payment.

As you can see, the refund that is correctly created for the order, is linked to the original payment, so there's no apparent reason not to link the credit log entry event to the payment:

![Screenshot 2023-01-23 at 16 59 41@2x](https://user-images.githubusercontent.com/167946/214087257-9dd4d988-7863-400a-8810-f229f2011a08.png)

**Solidus Version**:
All

**To Reproduce**

First of all `bundle exec rake sandbox` and `bin/rails s`

1. Complete a payment on the frontend
2. Go to admin, capture the payment of the newly created order
3. Refund the payment.
4. No log entry event for the refund.

**Current behavior**

No log entry event logged on the original payment for its refund.

**Expected behavior**

A log entry event logged on the original payment for its refund.

**Screenshots**

See above.

**Comment by senemsoy** [comment]
@benjaminwil and I are working on this.

**Comment by kennyadsl** [comment]
Hello @senemsoy! Are you still working on this? I think we are facing this in the Stripe extension, and if you and @benjaminwil did find anything worth sharing, it would be great to know. 🙂

**Comment by benjaminwil** [comment]
@kennyadsl We were able to reproduce the issue but haven't had time to find the source. We just created a pull request with a pending feature test so someone else can pick this up if it's urgent. 👍

**Comment by rainerdema** [comment]
After a recent team conversation, we have decided to postpone addressing this issue until the implementation of the new admin, which is already underway.

The main reason for this decision is that the logs are being saved correctly with the related `Spree::Refund` source, but the current view[ only displays logs originating from Spree::Payment source type](https://github.com/solidusio/solidus/blob/474877aaad9139a0a9b8564a9e7279e5446bc12c/backend/app/views/spree/admin/payments/show.html.erb#L27).

In conclusion, the issue is primarily related to view presentation rather than data integrity.

The overhaul of the admin interface will provide an opportune time to enhance the view presentation of these logs, ensuring a more comprehensive and accurate representation and including details about the log origin.

cc @solidusio/core-team
